protocol/js/flow-types-chat.js

```diff
@@ -16,6 +16,60 @@ export type RPCError = {
   code: number,
   desc: string
 }
+export type WaitingHandlerType = (waiting: boolean, method: string, sessionID: number) => void
+
+// $FlowIssue we're calling an internal method on engine that's there just for us
+const engineRpcOutgoing = (...args) => engine._rpcOutgoing(...args)
```

Comment:
This is weird that this imports engine for this. Maybe it would be better to have engine be the first argument when you make rpc calls?

It's double weird when you think that importing engine has the side effect of creating a new engine instance. We should probably export a singleton constructor instead.

In the singleton constructor, it doesn't make sense to build the engine here, because that would mean that anytime you want to use types you are building an engine.

I know @gabriel has a couple thoughts about this too.